### PR TITLE
feature(networks): using the unlock rpc proxy

### PR DESCRIPTION
**Pull request by julien51**
# Description

This is the next step in using our "proxied" RPC provider.
I am still a bit on the fence regarding rate limiting and I worry that some of the public RPC endpoints might block us.

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: provider/src/index.ts [hunk 1]
```diff
@@ -6,12 +6,14 @@ const supportedNetworks = {
   '4': 'https://eth-rinkeby.alchemyapi.io/v2/n0NXRSZ9olpkJUPDLBC00Es75jaqysyT',
   '5': 'https://goerli.prylabs.net',
   '10': 'https://mainnet.optimism.io',
-  '56': 'https://bsc-dataseed.binance.org/',
+  '56': 'https://rpc.ankr.com/bsc',
   '100':
     'https://cool-empty-bird.xdai.quiknode.pro/4edba942fb43c718f24480484684e907fe3fe1d3/',
   '137':
     'https://snowy-weathered-waterfall.matic.quiknode.pro/5b11a0413a62a295070c0dfb25637d5f8c591aba/',
+  '42161': 'https://rpc.ankr.com/arbitrum',
```
Comment (julien51) [inline provider/src/index.ts:14]: added Arbitrum and Avalanche in prep for launches...

**Comment by julien51** [comment]
> I assume you have tested it locally on dashboard and checkout to see the RPC calls are being forwarded correctly and does not introduce too much latency since there is proxy overhead.

Yes I did... but I will want to have this on staging for a few days before pushing to master because one user (me) might not make enough of a difference.

File: packages/networks/src/networks/rinkeby.ts [hunk 2]
```diff
@@ -2,9 +2,8 @@ import { NetworkConfig } from '@unlock-protocol/types'
 
 export const rinkeby: NetworkConfig = {
   publicProvider:
-    'https://eth-rinkeby.alchemyapi.io/v2/n0NXRSZ9olpkJUPDLBC00Es75jaqysyT',
-  provider:
-    'https://eth-rinkeby.alchemyapi.io/v2/n0NXRSZ9olpkJUPDLBC00Es75jaqysyT',
+    'https://rinkeby.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161',
```
[the remark this answers is not in the document]
Comment (julien51) [reply packages/networks/src/networks/rinkeby.ts:5]: No, the `publicProvider` is the one that can be added to the user's wallet when we prompt them to add a network. We should only add actual public providers (not our own RPC or private API keys). 
All the wallets I know of do come with settings for rinkeby and mainnet so I think that's ok.

**Comment by clemsos** [comment]
sidenote: will require to update the provider after merging Arbitrum and Avalanche #9221 #9224

**Pull request merged**